### PR TITLE
[Feature] Add `zoomToPage` on `CanvasController`

### PR DESCRIPTION
**Pull request by evan-mcgeek**
This PR adds a new functionality `zoomToPage()` via new CanvasController to be able to fit the page to the viewport OR the rectangle provided.

## Related tickets

-   [EDT-697](https://support.chili-publish.com/browse/EDT-697)

### CODE REVIEW

**Review by pkgacek — APPROVED** [review]

**Review by Dvergar — CHANGES_REQUESTED** [review]
LGTM but I requested a change and/or a feedback on engine-side which would have an effect on this PR, just blocking it to prevent a merge.

See: https://github.com/chili-publish/studio-engine/pull/36

File: src/controllers/ZoomController.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        pageId?: Id | null,
+        left?: number | null,
+        top?: number | null,
+        width?: number | null,
+        height?: number | null,
+    ) => {
+        const res = await this.#editorAPI;
+        return res.zoomToFit(pageId, left, top, width, height).then((result) => getEditorResponseData<null>(result));
+    };
+}
```
Comment (brapoprod) [inline src/controllers/ZoomController.ts:42]: Isn't it a bit overkill to have a zoomcontroller?
Doesn't it make sense to put all canvas related stuff under a CanvasController?
Maybe there we can make a distinction f.e.
SDK.canvas.zoom.zoomToFit?

Comment (Dvergar) [reply src/controllers/ZoomController.ts:42]: I'll leave @evan-mcgeek give its input but we discussed it several times and I think we agreed that there would be a discussion following the SDK PR.
We hesitated between putting it beside a zoom controller, canvas controller (or even a page controller).

I myself don't have a strong opinion, it's probably overkill for now, not sure how many zoom functions we are going to get (a second one is coming for general canvas zoom). Canvas controller looks the most generic.

_As a side-note, it's gonna be called zoomToPage._

@evan-mcgeek what do you think ?

Comment (evan-mcgeek) [reply src/controllers/ZoomController.ts:42]: I'd leave it separated IMO, considering some possible additional functionality in the future

Comment (evan-mcgeek) [reply src/controllers/ZoomController.ts:42]: so what would you say @brapoprod ?

**Review by brapoprod — APPROVED** [review]

**Review by Dvergar — APPROVED** [review]

**Pull request merged**